projects/core/src/cms/occ/occ-cms-component.adapter.ts

```diff
@@ -53,57 +46,83 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
     pageSize = ids.length,
     sort?: string
   ): Observable<CmsComponent[]> {
-    let requestParams = this.getRequestParams(pageContext, fields);
-    if (currentPage !== undefined) {
-      requestParams === ''
-        ? (requestParams = requestParams + 'currentPage=' + currentPage)
-        : (requestParams = requestParams + '&currentPage=' + currentPage);
-    }
-    if (pageSize !== undefined) {
-      requestParams = requestParams + '&pageSize=' + pageSize;
-    }
-    if (sort !== undefined) {
-      requestParams = requestParams + '&sort=' + sort;
-    }
+    console.log('second');
+
+    const requestParams = this.getComponentsRequestParams(
+      pageContext,
+      currentPage,
+      pageSize,
+      sort
+    );
 
     const idList: IdList = { idList: ids };
 
     return this.http
-      .post<CmsComponentList>(this.getBaseEndPoint() + `/components`, idList, {
-        headers: this.headers,
-        params: new HttpParams({
-          fromString: requestParams,
-        }),
-      })
+      .post<CmsComponentList>(
+        this.getComponentsEndpoint(requestParams, fields),
+        idList,
+        {
+          headers: this.headers,
+        }
+      )
       .pipe(
         pluck('component'),
         this.converter.pipeable(CMS_COMPONENT_LIST_NORMALIZER)
       );
   }
 
-  private getRequestParams(pageContext: PageContext, fields?: string): string {
-    let requestParams = '';
+  protected getComponentEndPoint(id: string, pageContext: PageContext): string {
```

Comment:
I'm getting a bit more conservative about making our API too complex to maintain. Can we make these methods private so we can change over time?

Reply:
well, maybe only `getComponentsRequestParams`. The others are quite useful.

Reply:
I've privatized a few functions to simplify things